doc/release-notes.md

```diff
@@ -4,3 +4,40 @@ release-notes at release time)
 Notable changes
 ===============
 
+Disabling old Sprout proofs
+---------------------------
+
+As part of our ongoing work to clean up the codebase and minimise the security
+surface of `zcashd`, we are removing `libsnark` from the codebase, and dropping
+support for creating and verifying old Sprout proofs. Funds stored in Sprout
+addresses are not affected, as they are spent using the hybrid Sprout circuit
+(built using `bellman`) that was deployed during the Sapling network upgrade.
+
+This change has several implications:
+
+- `zcashd` no longer verifies old Sprout proofs, and will instead assume they
+  are valid. This has a minor implication for nodes: during initial block
+  download, an adversary could feed the node fake blocks containing invalid old
+  Sprout proofs, and the node would accept the fake chain as valid. However,
+  `zcashd` internally contains checkpoints after Sapling activation for both
+  block heights and cumulative chain work, and does not exit the initial block
+  download phase until the active chain contains at least as much work as the
+  checkpointed chain work. The node would therefore be non-functional (and would
+  not broadcast the fake chain to other peers) until the fake chain contained as
+  much work as the main chain, making this a 50% + 1 attack, which the current
+  consensus rules already does not protect against.
```

Comment:
s/does/do/

Reply:
Why not make it an additional condition of leaving initial block download that Sapling activates at the correct block hash?

Reply:
Interesting idea; I think this would definitely help. That might make it trickier for someone to create a new chain based on our code, but probably not impossible. And they could always just revert the commit. I'll give this a go.

Reply:
I've given this further thought in light of https://github.com/zcash/zcash/pull/4060#discussion_r326156191 and I've decided that failing the above check should result in the node aborting. If we have found a chain with sufficient minimum chain work (assuming that `nMinimumChainWork` is set to at least the chain work of the most recently-activated NU hash we hard-code), and it does _not_ match the historic NU activation block hashes, this is most likely an adversarial chain.